genesyscloud/team/genesyscloud_team_proxy.go

```diff
@@ -0,0 +1,174 @@
+func newTeamProxy(clientConfig *platformclientv2.Configuration) *teamProxy {
+	api := platformclientv2.NewTeamsApiWithConfig(clientConfig)
+	return &teamProxy{
+		clientConfig:        clientConfig,
+		teamsApi:            api,
+		createTeamAttr:      createTeamFn,
+		getAllTeamAttr:      getAllTeamFn,
+		getTeamIdByNameAttr: getTeamIdByNameFn,
+		getTeamByIdAttr:     getTeamByIdFn,
+		updateTeamAttr:      updateTeamFn,
+		deleteTeamAttr:      deleteTeamFn,
+	}
+}
+
+// getTeamProxy acts as a singleton to for the internalProxy.  It also ensures
+// that we can still proxy our tests by directly setting internalProxy package variable
+func getTeamProxy(clientConfig *platformclientv2.Configuration) *teamProxy {
+	if internalProxy == nil {
+		internalProxy = newTeamProxy(clientConfig)
+	}
+
+	return internalProxy
+}
+
+// createTeam creates a Genesys Cloud team
+func (p *teamProxy) createTeam(ctx context.Context, team *platformclientv2.Team) (*platformclientv2.Team, error) {
+	return p.createTeamAttr(ctx, p, team)
+}
+
+// getTeam retrieves all Genesys Cloud team
+func (p *teamProxy) getAllTeam(ctx context.Context) (*[]platformclientv2.Team, error) {
+	return p.getAllTeamAttr(ctx, p)
+}
+
+// getTeamIdByName returns a single Genesys Cloud team by a name
+func (p *teamProxy) getTeamIdByName(ctx context.Context, name string) (id string, retryable bool, err error) {
+	return p.getTeamIdByNameAttr(ctx, p, name)
+}
+
+// getTeamById returns a single Genesys Cloud team by Id
+func (p *teamProxy) getTeamById(ctx context.Context, id string) (team *platformclientv2.Team, statusCode int, err error) {
+	return p.getTeamByIdAttr(ctx, p, id)
+}
+
+// updateTeam updates a Genesys Cloud team
+func (p *teamProxy) updateTeam(ctx context.Context, id string, team *platformclientv2.Team) (*platformclientv2.Team, error) {
+	return p.updateTeamAttr(ctx, p, id, team)
+}
+
+// deleteTeam deletes a Genesys Cloud team by Id
+func (p *teamProxy) deleteTeam(ctx context.Context, id string) (statusCode int, err error) {
+	return p.deleteTeamAttr(ctx, p, id)
+}
+
+// createTeamFn is an implementation function for creating a Genesys Cloud team
+func createTeamFn(ctx context.Context, p *teamProxy, team *platformclientv2.Team) (*platformclientv2.Team, error) {
+	team, _, err := p.teamsApi.PostTeams(*team)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create team: %s", err)
+	}
+
+	return team, nil
+}
+
+// getAllTeamFn is the implementation for retrieving all team in Genesys Cloud
+func getAllTeamFn(ctx context.Context, p *teamProxy) (*[]platformclientv2.Team, error) {
+	var allTeams []platformclientv2.Team
+	const pageSize = 100
+
+	teams, _, err := p.teamsApi.GetTeams(100, "", "", "", "")
+
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get team: %v", err)
+	}
+	if teams.Entities == nil || len(*teams.Entities) == 0 {
+		return &allTeams, nil
+	}
+	for _, team := range *teams.Entities {
+
+		allTeams = append(allTeams, team)
+	}
+
+	return &allTeams, nil
+}
```

Comment:
This function doesn't seem to get the next page of team objects, it only receives the first. There should be some sort of loop to go through every page until we hit an empty one

Reply:
comment addressed but it has not updated here.